lib/segment/src/data_types/vectors.rs

```diff
@@ -41,6 +42,19 @@ impl<'a> From<VectorRef<'a>> for &'a [VectorElementType] {
     }
 }
 
+impl<'a> TryFrom<VectorRef<'a>> for &'a SparseVector {
+    type Error = OperationError;
+
+    fn try_from(value: VectorRef<'a>) -> Result<Self, Self::Error> {
+        match value {
+            VectorRef::Dense(_) => Err(OperationError::service_error(
```

Comment:
I don't think it is a service error. It may be the wrong input.I prefer to add new error variant with wrong conversion between sparse and dense vectors